pkg/operator/connectivitycheckcontroller/connectivity_check_controller.go

```diff
@@ -111,18 +112,46 @@ func (c *connectivityCheckController) Sync(ctx context.Context, syncContext fact
 		return err
 	}
 
-	_, getCRDErr := c.apiextensionsClient.ApiextensionsV1().CustomResourceDefinitions().Get(ctx, podnetworkconnectivitychecksCRDName, metav1.GetOptions{})
+	crd, getCRDErr := c.apiextensionsClient.ApiextensionsV1().CustomResourceDefinitions().Get(ctx, podnetworkconnectivitychecksCRDName, metav1.GetOptions{})
 	if getCRDErr != nil && !errors.IsNotFound(getCRDErr) {
 		return getCRDErr
 	}
 
 	if !enabled {
 		// controller is not enabled
 		if errors.IsNotFound(getCRDErr) {
+			// crd has already been removed
 			return nil
 		}
+
+		// delete all podnetworkconnectivitychecks managed by this controller instance
+		checks, err := c.operatorcontrolplaneClient.ControlplaneV1alpha1().PodNetworkConnectivityChecks(c.namespace).List(ctx, metav1.ListOptions{})
+		if err != nil {
+			return err
+		}
+		for _, check := range checks.Items {
+			err := c.operatorcontrolplaneClient.ControlplaneV1alpha1().PodNetworkConnectivityChecks(c.namespace).Delete(ctx, check.Name, metav1.DeleteOptions{})
+			if err != nil {
+				return err
+			}
+		}
+
+		// if there are still podnetworkconnectivitychecks resources in other namespaces, do not delete the crd
+		checks, err = c.operatorcontrolplaneClient.ControlplaneV1alpha1().PodNetworkConnectivityChecks("").List(ctx, metav1.ListOptions{})
+		if err != nil {
+			return err
+		}
+		if len(checks.Items) > 0 {
+			return nil
+		}
+
+		// if podnetworkconnectivitycheck crd is less than 2 minutes old, don't delete yet to prevent racing with another instance
```

Comment:
this is actually to allow racing.  It gives two minutes for the other operator to add network check instances.  This is ok by me, but please create a better comment.

Reply:
Updated.